cache/redis_cache.go

```diff
@@ -106,76 +98,222 @@ func (r *redisCache) Get(key *Key) (*CachedData, error) {
 		log.Errorf("failed to get key %s with error: %s", key.String(), err)
 		return nil, ErrMissing
 	}
-
-	var payload redisCachePayload
-	err = json.Unmarshal([]byte(val), &payload)
-
+	ttl, err := r.client.TTL(ctx, key.String()).Result()
 	if err != nil {
-		log.Errorf("corrupted payload for key %s with error: %s", key.String(), err)
-		return nil, ErrMissing
+		return nil, fmt.Errorf("failed to ttl of key %s with error: %w", key.String(), err)
 	}
+	content, reader := r.fromByte([]byte(val))
 
-	ttl, err := r.client.TTL(ctx, key.String()).Result()
+	reader2 := &io_reader_decorator{Reader: reader}
+	value := &CachedData{
+		ContentMetadata: *content,
+		Data:            reader2,
+		Ttl:             ttl,
+	}
 
-	if err != nil {
-		log.Errorf("Not able to fetch TTL for: %s ", key)
+	return value, nil
+}
+*/
+
+func (r *redisCache) Get(key *Key) (*CachedData, error) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), getTimeout)
+	defer cancelFunc()
+	nbBytesToFetch := int64(100 * 1024)
+	// fetching 100kBytes from redis to be sure to have the full metadata and,
+	//  for most of the queries that fetch a few data, the cached results
+	val, err := r.client.GetRange(ctx, key.String(), 0, nbBytesToFetch).Result()
+	// if key not found in cache
+	if errors.Is(err, redis.Nil) || len(val) == 0 {
+		return nil, ErrMissing
 	}
 
-	decoded, err := base64.StdEncoding.DecodeString(payload.Payload)
+	// others errors, such as timeouts
 	if err != nil {
-		log.Errorf("failed to decode payload: %s , due to: %v ", payload.Payload, err)
+		log.Errorf("failed to get key %s with error: %s", key.String(), err)
 		return nil, ErrMissing
 	}
+	stringKey := key.String()
+	ttl, err := r.client.TTL(ctx, stringKey).Result()
```

Comment:
Isn't it better to execute it before `GetRange`? If it fails, we don't use fetched 100kb anyway

Reply:
I did that because it was the previous order (GET then TTL).
To be honest I don't think this change is need. if the TTL command starts to fail after a successful GETRANGE and it's not a one time failure then we will have a bigger problems to deal with